Clustering/Hdbscan/src/main/java/org/tribuo/clustering/hdbscan/HdbscanTrainer.java

```diff
@@ -147,10 +148,14 @@ public HdbscanModel train(Dataset<ClusterID> examples, Map<String, Provenance> r
         }
         ImmutableFeatureMap featureMap = examples.getFeatureIDMap();
 
-        SGDVector[] data = new DenseVector[examples.size()];
+        SGDVector[] data = new SGDVector[examples.size()];
         int n = 0;
         for (Example<ClusterID> example : examples) {
-            data[n] = DenseVector.createDenseVector(example, featureMap, false);
+            if (example.size() == featureMap.size()) {
```

Comment:
I believe all the unit tests will currently have `example.size() == featureMap.size()`. Is there any value to add a tests which provides sparse vectors?

Reply:
We usually run a sparse and a dense dataset through to make sure nothing breaks, but those tests have been there for a long time and are probably less valuable now as the vector creation infrastructure is more universally used. I probably should add one in this case though as a smoke test to make sure it executes.

Reply:
I added the small sparse data test that we use in `KMeans`.